pyquil/api/_qpu.py

```diff
@@ -154,3 +182,63 @@ def _build_patch_values(self) -> dict:
             patch_table[k.name][k.index] = v
 
         return patch_table
+
+    def _build_arithmetic_patch_values(self) -> str:
+        """
+        Update the variables shim with the final values to be patched into the gate parameters,
+        according to the arithmetic expressions in the original program.
+
+        For example:
+
+            DECLARE theta REAL
+            RZ(3 * theta) 0
+
+        gets translated to:
+
+            DECLARE theta REAL
+            DECLARE __P REAL[0]
+            RZ(__P[0]) 0
+
+        and the recalculation table gets the entry: __P[0] = '3 * theta'. In our example, we
```

Comment:
write this more pythony:

> `recalculation_table` is set to `{MemoryReference('__P', 0): "3 * theta", MemoryReference('__P', 1): "5 * theta"}`

Reply:
is it always going to be `"__P"`? why do we need to tag this along

Reply:
It's actually __P<random fixnum>. It can be whatever, I guess, whats a good way to describe it without giving it a name?

Reply:
__P < random fixnum >

Reply:
asked another way: is the return value of `_build_arithmetic_patch_values` always going to be `"__P"`?

Reply:
when I type `__P<randomfixnum>` it comes out as __P<randomfixnum>